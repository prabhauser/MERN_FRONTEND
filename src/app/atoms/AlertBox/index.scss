@import '../../../utilities/theme/_theme.scss';

.adbAlertbox {
    .cancel {
        display: flex;
        align-items: flex-end;
    }
    .MuiButtonBase-root {
        display: none;
    }

}
.alertDialog .alertImg {
    height: 100px;
    text-align: center !important;
}
.alertDialog .cancle {
    margin-left: 220px !important;
    margin-top: 13px !important;
}
.alertDialog .text {
    color: $s900Bg;
    font-family: Poppins;
    font-size: 17px;
    font-weight: 600;
    letter-spacing: 0;
}
.alertDialog .textValue {
    color: $p700Bg;
    font-family: Poppins;
    font-size: 16px;
    font-weight: 500;
    letter-spacing: 0;
}
.alertDialog .textHeader{
        // height: 28px;
        // width: 487px;
        color: $s900Bg;
        font-family: Poppins;
        font-size: 17px;
        font-weight: 600;
        letter-spacing: 0;
        line-height: 30px;
        text-align: center;
}
.alertDialog .changedText {
    // margin-left: 40px !important;
    text-align: center;
}

.MuiDialogContent-root {
    border-top: none !important;
    border-bottom: none !important;
}
.alertDialog .button {
    justify-content: center !important;
    display: flex !important;
    margin-top: 5px;
    // height: 44px;
    margin-bottom: 30px !important;
}
.alertDialog .leftButton {
    height: 44px;
    height: 46px;
    width: 178px;
    .MuiButton-label{
        height: 19px;
    width: 91px;
    color:$p800Bg;
    font-family: Poppins !important;
    font-size: 14px;
    font-weight: 600;
    letter-spacing: 0;
    line-height: 21px;
    text-align: center;
    }
}
.alertDialog .rightButton {
    height: 44px;
    height: 46px;
    width: 178px;
    
}
.MuiDialogActions-root {
    box-sizing: unset !important;
}
.alertDialog .textHolder {
    height: 28px;
    width: 487px;
    color: $s900Bg;
    font-family: Poppins;
    font-size: 17px;
    font-weight: 600;
    letter-spacing: 0;
    line-height: 30px;
    text-align: center;
    margin-left: 97px !important;
}
.alertDialog .testDev {
    margin-left: 97px !important;
    text-align: center !important;
}
.alertDialog  .cancel {
    text-align: end !important;
    cursor:pointer;
}
.alertDialog .remarkBox{
    margin:0 auto;
    width:77%;
}
.alertDialog .remarkHeader{
    text-align: center;
    color: #19253A;
  font-family: Poppins;
  font-size: 20px;
  font-weight: 600;
  letter-spacing: 0;
  line-height: 30px;
}
.alertDialog .remarkBox .label-style{
    opacity: 0.6;
  color: #02122C;
  font-family: Poppins;
  font-size: 14px;
  font-weight: 500;
  letter-spacing: 0;
}
.alertDialog .remarkBox textarea{
    margin:0 auto;
    width:100%;
    height:160px;
    border-radius:10px;
  border: 1px solid #CCD0D5;
  border-radius: 10px;
  background-color: #FCFDFF
}

@media (max-width: 420px) {
    // .textValue {
    //     margin-left: -20px !important;
    // }
    // .changedText {
    //     margin-left: 30px;
    //     display: inline !important;
    // }
    .alertDialog .textHolder {
        margin-left: 50px !important;
        display: table-row !important;
    }
    .alertDialog .leftButton {
        width: 112px !important;
    }
    .alertDialog .rightButton {
        width: 112px !important;
    }
}

@media (min-width: 421px) and (max-width: 540px) {
    .alertDialog .textHolder {
        margin-left: 45px !important;
    }
    .alertDialog .changedText {
        margin-left: 45px !important;
        display: contents !important;
    }
}
