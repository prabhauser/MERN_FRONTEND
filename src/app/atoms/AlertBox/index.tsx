import React from 'react';
import { withStyles, makeStyles } from '@material-ui/core/styles';
import { unit } from '../../constants/constants';
import { TextField, Box } from '@mui/material';
import themeColors from '../../../utilities/theme/theme.module.scss';
import { Dialog, Button } from '@material-ui/core';
import MuiDialogTitle from '@material-ui/core/DialogTitle';
import MuiDialogContent from '@material-ui/core/DialogContent';
import MuiDialogActions from '@material-ui/core/DialogActions';
import IconButton from '@material-ui/core/IconButton';
import CloseIcon from '@material-ui/icons/Close';
import Typography from '@material-ui/core/Typography';
import ADBButton from '../Button';
import closeIcon from '../../../assets/images/close-icon.svg';
import alertIcon from '../../../assets/images/alertIcon.svg';
import Label from '../Label/index';
import './index.scss';

type ComponentProps = {
    handleClose: React.MouseEventHandler<HTMLElement>;
    isOpen: boolean;
    leftBtnPress?: React.MouseEventHandler<HTMLButtonElement>;
    rightBtnPress?: React.MouseEventHandler<HTMLButtonElement>;
    entityDetails?: EntityDetails;
    modalContent?: any;
    modalHeader?: any;
    remark?: boolean;
    validateRemark?: any;
};
type EntityDetails = {
    entityName: string;
    entityOld: string;
    entityNew: string;
};

const useStyles = makeStyles({
    root: {
        margin: 0,
        padding: `${unit.spacing * 2}px`,
        '& .MuiDialog-paperWidthSm': {
            background: '#F1F3F6'
        }
    },
    closeButton: {
        position: 'absolute',
        right: `${unit.spacing}px`,
        top: `${unit.spacing}px`,
        color: `${themeColors.s100Bg}`
    },
    modalHeader: {
        fontSize: '22px'
    }
});

type DialogTitleType = {
    id: string;
    children?: any;
    classes?: any;
    onClose?: React.MouseEventHandler<HTMLElement>;
};

const DialogTitle = (props: DialogTitleType) => {
    const { children, classes, onClose, ...other } = props;
    return (
        <MuiDialogTitle disableTypography className={classes?.root} {...other}>
            <Typography>{children}</Typography>
            <div className="alertImg">
                <img src={alertIcon} className="alertImg" />
            </div>
        </MuiDialogTitle>
    );
};

const DialogContent = withStyles((theme) => ({
    root: {
        padding: theme.spacing(2)
    }
}))(MuiDialogContent);

const DialogActions = withStyles((theme) => ({
    root: {
        margin: 0,
        padding: theme.spacing(1)
    }
}))(MuiDialogActions);

const AlertBox = (props: ComponentProps) => {
    const classesList = useStyles();
    const {
        handleClose,
        isOpen,
        leftBtnPress,
        rightBtnPress,
        entityDetails,
        modalContent,
        modalHeader,
        remark,
        validateRemark
    } = props;
    return (
        <div className="adbAlertbox">
            <Dialog
                PaperProps={{
                    // style: { borderRadius: 40, height: 458, width: 722 }
                    style: { borderRadius: 40, width: 722 }
                }}
                onClose={handleClose}
                aria-labelledby="customized-dialog-title"
                open={isOpen}
                className="alertDialog"
            >
                <div className="cancel">
                    <DialogTitle id="customized-dialog-title" onClose={handleClose}>
                        <img onClick={handleClose} src={closeIcon} />
                    </DialogTitle>
                </div>

                {modalContent ? (
                    <DialogContent dividers>
                        <Typography component={'span'} gutterBottom>
                            <div className="changedText">
                                <span
                                    className="textHeader"
                                    style={{ marginLeft: '0px !important', marginBottom: '20px' }}
                                >
                                    {modalHeader}
                                </span>
                                <br />
                                <span className="textValue coloredText" style={{ color: 'lightgrey' }}>
                                    {modalContent}
                                </span>
                            </div>
                        </Typography>
                    </DialogContent>
                ) : remark ? (
                    <Box component="form" noValidate autoComplete="off">
                        <h4 className="remarkHeader">Please type your remark to reject this request</h4>
                        <div className="remarkBox">
                            <Label label="Remarks" />
                            <textarea onChange={(ele) => validateRemark(ele)} />
                        </div>
                    </Box>
                ) : (
                    <DialogContent dividers>
                        <Typography component={'span'} gutterBottom>
                            <span className="textHolder">
                                Are you sure you want to change {entityDetails?.entityName}?
                            </span>
                            <br />
                            <br />
                            <div className="changedText">
                                <span className="textValue">
                                    You are about to change the {entityDetails?.entityName} from{' '}
                                </span>
                                <span className="text">
                                    {entityDetails?.entityOld} to {entityDetails?.entityNew}
                                </span>
                                <br />
                            </div>
                        </Typography>
                    </DialogContent>
                )}

                <DialogActions className="button">
                    <ADBButton
                        name={'No'}
                        handleClick={leftBtnPress}
                        secondary={true}
                        color="primary"
                        type="submit"
                        className="leftButton"
                    />
                    <ADBButton
                        name={'Yes'}
                        handleClick={rightBtnPress}
                        color="primary"
                        type="submit"
                        className="rightButton"
                    />
                </DialogActions>
            </Dialog>
        </div>
    );
};

export default AlertBox;
