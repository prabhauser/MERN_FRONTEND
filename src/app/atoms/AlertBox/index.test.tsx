import { shallow } from 'enzyme';
import AlertBox from './index';

describe('AlertBox Component', () => {
    it('should take AlertBox snapshot', () => {
        const component = shallow(
            <AlertBox
                leftBtnPress={() => {}}
                rightBtnPress={() => {}}
                entityDetails={{ entityName: 'test', entityOld: 'test', entityNew: 'test' }}
                handleClose={() => {}}
                isOpen={false}
            />
        );
        expect(component).toMatchSnapshot();
    });
});
