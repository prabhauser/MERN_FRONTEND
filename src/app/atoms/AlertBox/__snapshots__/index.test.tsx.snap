// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlertBox Component should take AlertBox snapshot 1`] = `
<div
  className="adbAlertbox"
>
  <WithStyles(ForwardRef(Dialog))
    PaperProps={
      Object {
        "style": Object {
          "borderRadius": 40,
          "width": 722,
        },
      }
    }
    aria-labelledby="customized-dialog-title"
    className="alertDialog"
    onClose={[Function]}
    open={false}
  >
    <div
      className="cancel"
    >
      <DialogTitle
        id="customized-dialog-title"
        onClose={[Function]}
      >
        <img
          onClick={[Function]}
          src="close-icon.svg"
        />
      </DialogTitle>
    </div>
    <WithStyles(WithStyles(ForwardRef(DialogContent)))
      dividers={true}
    >
      <WithStyles(ForwardRef(Typography))
        component="span"
        gutterBottom={true}
      >
        <span
          className="textHolder"
        >
          Are you sure you want to change 
          test
          ?
        </span>
        <br />
        <br />
        <div
          className="changedText"
        >
          <span
            className="textValue"
          >
            You are about to change the 
            test
             from
             
          </span>
          <span
            className="text"
          >
            test
             to 
            test
          </span>
          <br />
        </div>
      </WithStyles(ForwardRef(Typography))>
    </WithStyles(WithStyles(ForwardRef(DialogContent)))>
    <WithStyles(WithStyles(ForwardRef(DialogActions)))
      className="button"
    >
      <ADBButton
        className="leftButton"
        color="primary"
        handleClick={[Function]}
        name="No"
        secondary={true}
        type="submit"
      />
      <ADBButton
        className="rightButton"
        color="primary"
        handleClick={[Function]}
        name="Yes"
        type="submit"
      />
    </WithStyles(WithStyles(ForwardRef(DialogActions)))>
  </WithStyles(ForwardRef(Dialog))>
</div>
`;
